Make the `getMessage` function for the javascript example more robust. (#39672)
mdn/content: files/en-us/mozilla/add-ons/webextensions/native_messaging/index.md

import fs from "node:fs/promises";

let readPromise = null; // Prevents reading until an in-progress read has finished.

> // This function cannot be called multiple times in parallel because they
> // share access to /dev/stdin. We use a lock `readPromise` to prevent this.

Reviewer: Perhaps I'm not fully understanding this comment, but it seems as though it should read like this:
```suggestion
// This function should not be called in parallel, because each call would
// share access to /dev/stdin. A `readPromise` lock is used to prevent this.
```

import fs from "node:fs/promises";

> const _r = rs=>_rs_rp = rs; let _rs_rp, _rp = null; // read promise

Author: I prefer the one liner. If you don't like it, then change it to whatever it needs to be. I don't know how to do that.

async function getMessage() {
  while (_rp) await _rp; // wait for previous reads
  _rp = new Promise(_r); // make any further calls to getMessage wait
  const input = await fs.open("/dev/stdin");
  const header = new Uint32Array(1);
  await input.read(header);
  const message = new Uint8Array(header[0]);
  await input.read(message);
  await input.close();

>   _rs_rp(); _rp = null; // allow the next getMessage call to proceed

Author: I prefer the one liner. I resolve the promise, then I set it to null. In my head, it makes sense to be on one line, but if you don't like it then change it. Thank you.
